src/aliceVision/mesh/Mesh.cpp

```diff
@@ -2258,284 +2265,151 @@ void Mesh::getLargestConnectedComponentTrisIds(StaticVector<int>& out) const
-                        vertex2.y = vertex.z; // 3rd point of the 1st triangle is the 2nd of the 2nd triangle.
-                        uvCoord2.y = uvCoord.z;
-                        ok = true;
-                        withUV = true;
-                        withQuad = true;
-                    }
-                    else if(n1 == 8)
-                    {
-                        sscanf(line.c_str(), "f %i/%i/%i %i/%i/%i %i/%i/%i %i/%i/%i",
-                               &vertex.x, &uvCoord.x, &vertexNormal.x,
-                               &vertex.y, &uvCoord.y, &vertexNormal.y,
-                               &vertex.z, &uvCoord.z, &vertexNormal.z,
-                               &vertex2.z, &uvCoord2.z, &vertexNormal2.z);
-                        vertex2.x = vertex.x; // same first point
-                        uvCoord2.x = uvCoord.x;
-                        vertexNormal2.x = vertexNormal.x;
-                        vertex2.y = vertex.z; // 3rd point of the 1st triangle is the 2nd of the 2nd triangle.
-                        uvCoord2.y = uvCoord.z;
-                        vertexNormal2.y = vertexNormal.z;
-                        ok = true;
-                        withUV = true;
-                        withNormal = true;
-                        withQuad = true;
-                    }
+                    aiColor4D c = mesh->mColors[0][idPoint];
+                    double r = c.r * 255.0;
+                    double g = c.g * 255.0;
+                    double b = c.b * 255.0;
+                    _colors.push_back(aliceVision::rgb(r, g, b));
                 }
-                else
+
+                if (mesh->HasTextureCoords(0))
                 {
-                    if(n2 == 3)
-                    {
-                        sscanf(line.c_str(), "f %i//%i %i//%i %i//%i", &vertex.x, &vertexNormal.x, &vertex.y, &vertexNormal.y, &vertex.z, &vertexNormal.z);
-                        ok = true;
-                        withNormal = true;
-                    }
-                    else if(n2 == 4)
-                    {
-                        sscanf(line.c_str(), "f %i//%i %i//%i %i//%i %i//%i",
-                               &vertex.x, &vertexNormal.x,
-                               &vertex.y, &vertexNormal.y,
-                               &vertex.z, &vertexNormal.z,
-                               &vertex2.z, &vertexNormal2.z);
-                        vertex2.x = vertex.x; // same first point
-                        vertexNormal2.x = vertexNormal.x;
-                        vertex2.y = vertex.z; // 3rd point of the 1st triangle is the 2nd of the 2nd triangle.
-                        vertexNormal2.y = vertexNormal.z;
-                        ok = true;
-                        withNormal = true;
-                        withQuad = true;
-                    }
+                    aiVector3D t = mesh->mTextureCoords[0][idPoint];
+                    uvCoords.push_back(Point2d(t.x, t.y));
```

Comment:
```suggestion
                    uvCoords.emplace_back(t.x, t.y);
```
requires adding the method to `StaticVector`